### ISSUE TITLE
Investigate filtering by Taxonomy results

### ISSUE THREAD

**Issue by bognasoldev89** [benetech/ServiceNet#789]
On staging environment, I set "Work Clothing" Taxonomy filter and unchecked "Only show matching"

![789 1](https://user-images.githubusercontent.com/51324797/72417761-dad23080-3779-11ea-9078-382e859674f3.png)

And I got 4 hits:

![789 2](https://user-images.githubusercontent.com/51324797/72417797-ee7d9700-3779-11ea-931e-8f2cee76fa6f.png)

The issue is, Dress for Success San Francisco has a different taxonomy: Job Finding Assistance.
And while setting a Taxonomy filter for Job Finding Assistance I got 21 hits.

![789 3](https://user-images.githubusercontent.com/51324797/72418056-664bc180-377a-11ea-9990-b7cbaa2b185c.png)

I don't get logic besides this. Is there any relation between those taxonomies?
That's the one example, I think I could provide more if needed.

**Comment by jkondrat** [comment]
@bognasoldev89 @johnhbenetech : It seems that the taxonomy filter works well, the issue is that we are only showing the first taxonomy on multi-record-view (as a simple text field). Should we show all of them? If so, how should it look like?

**Comment by johnhbenetech** [comment]
@jkondrat 
What about something like this, one pill per taxonomy term
![screenshot of UI featuring multiple taxonomies](https://user-images.githubusercontent.com/12927553/73963741-7e74b380-48d6-11ea-8723-fdc08fcc9f88.png)

**Comment by jkondrat** [comment]
@johnhbenetech : Is it okay to use bootstrap pills? https://getbootstrap.com/docs/4.4/components/badge/#pill-badges
![Screenshot from 2020-02-07 13-30-20](https://user-images.githubusercontent.com/5969801/74029981-3628c480-49ae-11ea-9278-15f850a5c4fb.png)
![Screenshot from 2020-02-07 13-31-05](https://user-images.githubusercontent.com/5969801/74029986-36c15b00-49ae-11ea-8566-480b1255acc2.png)
![Screenshot from 2020-02-07 13-31-26](https://user-images.githubusercontent.com/5969801/74029988-36c15b00-49ae-11ea-91fe-c55b9e80c7e6.png)

**Comment by johnhbenetech** [comment]
@jkondrat Seems good to me!

**Comment by bognasoldev89** [comment]
I checked this for all of the providers and it looks fabulous:

![Screenshot from 2020-02-11 16-03-55](https://user-images.githubusercontent.com/51324797/74249686-d7cf4f00-4ce9-11ea-9c80-e2f790fdc4a8.png)

![Screenshot from 2020-02-11 16-14-55](https://user-images.githubusercontent.com/51324797/74249687-d867e580-4ce9-11ea-88f5-d93fdd79944f.png)

I stomped on something like that though (HOPE REHABILITATION SERVICES):

![Screenshot from 2020-02-11 16-11-55](https://user-images.githubusercontent.com/51324797/74249909-2bda3380-4cea-11ea-98ec-73e25c68e69f.png)

As far as I am informed, it's taxonomy code and it's displayed because this taxonomy lacks name. @johnhbenetech should it be displayed this way or maybe hidden? I believe it's not a common case.
